src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -535,7 +538,31 @@ public Response getVersionMetadataBlock( @PathParam("id") String datasetId,
             return notFound("metadata block named " + blockName + " not found");
         });
     }
-    
+
+    /**
+     * Add Signposting
+     * @param datasetId
+     * @param versionId
+     * @param uriInfo
+     * @param headers
+     * @return
+     */
+    @GET
+    @Path("{id}/versions/{versionId}/linkset")
+    public Response getLinkset( @PathParam("id") String datasetId, @PathParam("versionId") String versionId, @Context UriInfo uriInfo, @Context HttpHeaders headers) {
+        if ( ":draft".equals(versionId) ) {
+            return badRequest("The :draft version can be viewed");
+        }
+        return response( req -> {
+            DatasetVersion dsv = getDatasetVersionOrDie(req, versionId, findDatasetOrDie(datasetId), uriInfo, headers);
+            String dataverseSiteUrl = systemConfig.getDataverseSiteUrl();
+            String anchor = dataverseSiteUrl + "/dataset.xhtml?persistentId=" + dsv.getDataset().getPersistentURL();
+            String signpostingConf = settingsService.getValueForKey(SettingsServiceBean.Key.SignpostingConf, BundleUtil.getStringFromBundle("signposting.configuration.SignpostingConf"));
+            if (signpostingConf.isEmpty()) return notFound("Configuration key for signposting is empty [SignpostingConf]");
+            if (dsv.getId() == null) return notFound("Dataset not found: Id is empty");
```

Comment:
I'm wondering about the error responses - not sure I would want users to know whether the config key exists. One option would just be to send no message or "Signposting not enabled". I also wonder if 501/not implemented or 409/conflict be better for the first?
For the second - I assume the getDatasetVersionOrDie (findDatasetOrDie) above will return before you could hit this?